projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/dag_validator.py

```diff
@@ -141,6 +149,28 @@ def _validate_fail_meta_job_on_error(self, job: Dict):
                 f" is {type(job['fail_meta_job_on_error'])}. Expected type is bool.",
             )
 
+    def _validate_arguments(self, job: Dict):
+        if "arguments" in job:
+            if not isinstance(job["arguments"], Dict):
+                self._raise_error(
+                    list(job["job_name"]),
+                    ERROR.TYPE,
+                    "The type of the job dict key arguments is not str.",
+                    f"Change the Data Job Dict value of arguments. "
+                    f"Current type is {type(job['arguments'])}. Expected type is JSON-formatted str.",
+                )
```

Comment:
You are checking if job["arguments"} is Dict, but then saying it must be a string. 

It should be a python dictionary by the way. Not a string. It's a not a nice thing to force user to format their arguments as a string. In Python of course we should be using pythonic objects. 

Better validation is to check if it can be serialized as json. 

So as a validation running `json.dumps` and not loads .

Reply:
Makes sense, my bad, changed it.